### PR TITLE
feat: get images on main

### PR DESCRIPTION
**Pull request by i-chvets**
For more details refer to the following issue: https://github.com/canonical/bundle-kubeflow/issues/679

Summary of changes:
- Added script that produces list of container images managed by charm in this repository. Image list is a dynamic list.

NOTE: Script replaced outdated get-images-1.7-stable.sh script.

### CODE REVIEW

**Comment by orfeas-k** [comment]
Not sure if we should merge this since https://github.com/canonical/metacontroller-operator/issues/80. If I send a PR for this, then we should modify this script as well.

File: tools/get-images.sh
Comment (phoevos) [inline tools/get-images.sh:7]: Following the latest changes merged to `main`, exposing the oci-image in `metadata.yaml`, we will have to make the following change here:
```suggestion
IMAGE_LIST+=($(find -type f -name metadata.yaml -exec yq '.resources | to_entries | .[] | .value | ."upstream-source"' {} \;))
```

Comment (phoevos) [reply tools/get-images.sh:7]: What worries me though is that this script won't work for `track/2.0` (the equivalent applies to a couple of our other repos where we recently introduced configurable images). Should we backport this to `track/2.0`, keeping the logic of the initial `get-images-1.7-stable.sh` or do we not need it?

**Review by phoevos — CHANGES_REQUESTED** [review]
Minor change needed, taking the latest additions to `main` into account.

File: tools/get-images.sh
Comment (i-chvets) [reply tools/get-images.sh:7]: @phoevos Initial `get-images-1.7-stable.sh` was never triggered. `track/2.0` has its [own script](https://github.com/canonical/metacontroller-operator/blob/track/2.0/tools/get-images.sh). So, this script applies only to main, it is just git replaced script. If you think it is cleaner to remove non-used script first and then commit new script let's do this, but I think this PR is clear enough.
If you have no reservations please resolve and squash-and-merge this.

Comment (i-chvets) [reply tools/get-images.sh:7]: @phoevos Do not merge it just yet, I will need to address Orfeas concern, to make this script parse configurable images.

Comment (phoevos) [reply tools/get-images.sh:7]: Won't the above suggestion do?

Comment (i-chvets) [reply tools/get-images.sh:7]: Yes. It will just checked.

**Comment by i-chvets** [comment]
@phoevos Updated with your suggestion. Please approve and merge when ready.

**Review by phoevos — APPROVED** [review]

**Comment by phoevos** [comment]
CI is green, @i-chvets feel free to merge!

**Pull request merged**
